### PR TITLE
Remove graph config innards from defaultProps

### PR DESCRIPTION
**Pull request by alexcjohnson**
Addresses *most* of #514 - doesn't include a test (yet?), but removes the config innards and adds the missing pieces to `propTypes`.

### CODE REVIEW

File: test/test_integration.py [hunk 1]
```diff
@@ -1193,7 +1193,7 @@ def on_click(n_clicks):
     def check_plotlyjs(self):
         # find plotly.js files in the dist folder, check that there's only one
         all_dist = os.listdir(dcc.__path__[0])
-        js_re = '^plotly-(.*)\.min\.js$'
+        js_re = r'^plotly-(.*)\.min\.js$'
```
Comment (alexcjohnson) [inline test/test_integration.py]: oddly I can't get this error to appear locally... 😐

File: test/test_integration.py [hunk 2]
```diff
@@ -1184,6 +1185,54 @@ def on_click(n_clicks):
...
+        dist_version = re.match(js_re, plotlyjs_dist[0]).groups()[0]
+        self.assertEqual(page_version, dist_version)
+
+    def check_graph_config_shape(self):
+        config_schema = self.driver.execute_script(
+            'return Plotly.PlotSchema.get().config;'
+        )
+        with open(os.path.join(dcc.__path__[0], 'metadata.json')) as meta:
+            graph_meta = json.load(meta)['src/components/Graph.react.js']
+            config_prop_shape = graph_meta['props']['config']['type']['value']
```
Comment (alexcjohnson) [inline test/test_integration.py]: kind of ugly to have to use `metadata.json` for this... I was hoping I could get this directly from the `Graph` component, but `PropTypes` seems to turn everything into a validation function, I don't see a way to get the input params back out.

File: test/test_integration.py [hunk 3]
```diff
@@ -1184,6 +1185,54 @@ def on_click(n_clicks):
...
+
+        def crawl(schema, props):
+            for prop_name in props:
+                self.assertIn(prop_name, schema)
+
+            for item_name, item in schema.items():
+                if item_name in ignored_config:
+                    continue
+
+                self.assertIn(item_name, props)
```
Comment (alexcjohnson) [inline test/test_integration.py]: Currently not testing anything about the props, just that the right keys exist. We could enhance this later to verify that their types match what's in the schema. Note that we could have put back in the deep defaults and used a similar framework to test that they match the schema, but I don't think that's a good idea. We'd still have the problem of a single override dropping the defaults - in principle picking up the right defaults again from plotly.js, but there could be subtle problems with this, or perhaps a user overrides the plotly.js version... I'd rather keep the looser coupling.

File: test/test_integration.py [hunk 4]
```diff
@@ -1184,6 +1185,54 @@ def on_click(n_clicks):
...
+        self.check_graph_config_shape()
+        self.check_plotlyjs()
+
+    def check_plotlyjs(self):
+        # find plotly.js files in the dist folder, check that there's only one
+        all_dist = os.listdir(dcc.__path__[0])
+        js_re = r'^plotly-(.*)\.min\.js$'
+        plotlyjs_dist = [fn for fn in all_dist if re.match(js_re, fn)]
+
+        self.assertEqual(len(plotlyjs_dist), 1)
```
Comment (alexcjohnson) [inline test/test_integration.py]: I don't know how we ended up with 2 copies of plotly.js on master, 1.44.1 and 1.45.0... the associated PRs all seem to swap one file for another, maybe a mis-merge somewhere. But anyway this test will prevent that in the future.

File: src/components/Graph.react.js [hunk 5]
```diff
@@ -516,11 +533,43 @@ const graphPropTypes = {
...
+            scale: PropTypes.number,
+        }),
+
         /**
          * add the plotly logo on the end of the mode bar
          */
         displaylogo: PropTypes.bool,
 
+        /**
+         * add the plotly logo even with no modebar
```
Comment (Marc-Andre-Rivet) [inline src/components/Graph.react.js]:
```suggestion
         * Add the plotly logo even with no modebar
```

File: src/components/Graph.react.js [hunk 6]
```diff
@@ -491,6 +501,13 @@ const graphPropTypes = {
          */
         displayModeBar: PropTypes.oneOf([true, false, 'hover']),
 
+        /**
+         * should we include a modebar button to send this data to a
```
Comment (Marc-Andre-Rivet) [inline src/components/Graph.react.js]:
```suggestion
         * Should we include a modebar button to send this data to a
```

File: src/components/Graph.react.js [hunk 7]
```diff
@@ -374,6 +374,11 @@ const graphPropTypes = {
          */
         staticPlot: PropTypes.bool,
 
+        /**
+         * base URL for a Plotly cloud instance, if `showSendToCloud` is enabled
```
Comment (Marc-Andre-Rivet) [inline src/components/Graph.react.js]:
```suggestion
         * Base URL for a Plotly cloud instance, if `showSendToCloud` is enabled
```

File: src/components/Graph.react.js (hunk 5 above)
Comment (alexcjohnson) [reply src/components/Graph.react.js]: Ah right, a lot of the older docstrings didn't bother capitalizing so I tried to follow that, but then some of the newer ones do... I should just go through and capitalize them all. Thanks for pointing this out.

**Pull request merged**